### PR TITLE
feat: [CDS-56689]: Add rancher connector

### PR DESCRIPTION
**Pull request by abhinavsingh1196**
## Describe your changes

go-sdk entities change here: https://github.com/harness/harness-go-sdk/pull/347
Docs review ticket: https://harness.atlassian.net/browse/DOC-3342
Feature epic: https://harness.atlassian.net/browse/CDS-50562

## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`

### CODE REVIEW

**Review by rathodmeetsatish — DISMISSED** [review]

File: internal/service/platform/connector/k8s_rancher.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+				Optional:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"rancher_url": {
+							Description: "The URL of the Rancher cluster.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"password_ref": {
```
Comment (abosii-harness) [inline internal/service/platform/connector/k8s_rancher.go:43]: Should be this bearer_token_ref?

File: internal/service/platform/connector/k8s_rancher.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+			},
+
+			"bearer_token": {
+				Description: "URL and bearer token for the rancher cluster.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"rancher_url": {
```
Comment (abosii-harness) [inline internal/service/platform/connector/k8s_rancher.go:38]: Does it make sense to have rancher_url as part of the bearer_token?

**Review by abosii-harness — CHANGES_REQUESTED** [review]

File: internal/service/platform/connector/k8s_rancher.go (hunk 1 above)
Comment (abhinavsingh1196) [reply internal/service/platform/connector/k8s_rancher.go]: yes, renamed

File: internal/service/platform/connector/k8s_rancher.go (hunk 2 above)
Comment (abhinavsingh1196) [reply internal/service/platform/connector/k8s_rancher.go]: It makes more sense to keep it out. moved it, thanks

**Review by abosii-harness — APPROVED** [review]

File: .changelog/598.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+```release-note:new-resource
+harness_platform_connector_rancher - Added rancher connector in harness terraform provider
```
Comment (deepakbolangady) [inline .changelog/598.txt:2]: (Global suggestion)
```suggestion
harness_platform_connector_rancher - Added a Rancher connector to the Terraform Harness provider.
```
Is it "Harness Terraform provider" or "Terraform Harness provider"? Has Harness created a Terraform provider or has Terraform enabled Harness to create a provider? If we expand to "Harness provider for Terraform", we might get the answer: "Terraform Harness provider" sounds better, so I'm leaning toward the latter. 

For now, we can hold this thought and maybe just use uppercase "H" and "T" globally. Retain the case of "p" in "provider" (should be lowercase). I will do some research and come back on which one is the adjective. 

Rancher is a company/product. Use uppercase. Globally.

File: docs/data-sources/platform_connector_rancher.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "harness_platform_connector_rancher Data Source - terraform-provider-harness"
+subcategory: "Next Gen"
```
Comment (deepakbolangady) [inline docs/data-sources/platform_connector_rancher.md:4]: NextGen is one word. I know it cannot be done piecemeal. Can it be corrected globally sometime?

File: docs/data-sources/platform_connector_rancher.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+description: |-
+  Datasource for looking up a Rancher connector.
+---
+
+# harness_platform_connector_rancher (Data Source)
+
+Datasource for looking up a Rancher connector.
```
Comment (deepakbolangady) [inline docs/data-sources/platform_connector_rancher.md:11]:
```suggestion
Data source for looking up a Rancher connector.
```

File: docs/data-sources/platform_connector_rancher.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+## Example Usage
```
Comment (deepakbolangady) [inline docs/data-sources/platform_connector_rancher.md:13]:
```suggestion
## Example usage
```
(Use sentence case in titles. We do that in docs. If this is a heavy lift, set aside for now.)

**Review by rathodmeetsatish — DISMISSED** [review]

File: internal/service/platform/connector/k8s_rancher_data_source.go [hunk 7]
```diff
@@ -0,0 +1,46 @@
+package connector
+
+import (
+	"github.com/harness/terraform-provider-harness/helpers"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DatasourceConnectorRancher() *schema.Resource {
+	resource := &schema.Resource{
+		Description: "Datasource for looking up a Rancher connector.",
```
Comment (deepakbolangady) [inline internal/service/platform/connector/k8s_rancher_data_source.go:10]:
```suggestion
		Description: "Data source for looking up a Rancher connector.",
```
(Maybe check globally?)

**Review by deepakbolangady — DISMISSED** [review]
I've requested some minor changes. However, I don't want to block this PR, so I'm approving it. If this PR needs another approval from me, please reach out.

File: internal/service/platform/connector/k8s_rancher_data_source.go (hunk 7 above)
Comment (abhinavsingh1196) [reply internal/service/platform/connector/k8s_rancher_data_source.go]: Done

File: docs/data-sources/platform_connector_rancher.md (hunk 5 above)
Comment (abhinavsingh1196) [reply docs/data-sources/platform_connector_rancher.md]: Done

**Review by abosii-harness — APPROVED** [review]

**Review by rathodmeetsatish — APPROVED** [review]

**Pull request merged**